Tasks/Gordeenko/Task2/Task2.py

```diff
@@ -0,0 +1,139 @@
+        h2 = (int(h+1))
+        hour_3 = d.get(h2)
+        m3 = (int(60-m))
+        minute_3 = d_8.get(m3)
+        minute_4 = d_7.get(m3)
+        print(f"без {minute_3[0]} минут{minute_4[0]} {hour_3[0]}")
+    else: print("error")
+if x==1: # user 
+    hh,mm = input("Please enter the time in hour and minute format hh:mm - 4 numbers across (:) ").split(":")
+    h = int(hh)
+    m = int(mm)
+    if h>=0 and m == 0: # время ровно
+        hour = d.get(h)
+        hour_2 = d_2.get(h)
+        print(f"{hour[0]} час{hour_2[0]}")
+    elif (h>=12 and h<=20) and (m>=1 and m<=20 or m==30 or m==40): # (часы >=12 до 20 вкл   /   минуты 1-20,30,40)
+        hour = d.get(h)
+        hour_2 = d_2.get(h)
+        h2 = (int(h-1))
+        hour_3 = d_5.get((int(str(h2)[1])))
+        minute = d_3.get(m)
+        minute_2 = d_6.get(m)
+        print(f"{minute[0]} минут{minute_2[0]} {hour_3[0]}")
+    elif ((h>=0 and h<=11) or h>20) and (m<=20 or m==30 or m==40):  # (часы <12 и 21-23  /    минуты 1-20,30,40)
+        hour = d.get(h)
+        hour_2 = d_2.get(h)
+        h2 = (int(h+1))
+        hour_3 = d_5.get(h2)
+        minute = d_3.get(m)
+        minute_2 = d_6.get(m)
+        print(f"{minute[0]} минут{minute_2[0]} {hour_3[0]}")
+    elif (h>=12 and h<=20) and ((m>=21 and m<=29) or (m>=31 and m<=39) or (m>=41 and m<=44)):  # (часы >=12 до 20 вкл  /    минуты 21-29, 31-39, 41-44)
+        hour = d.get(h)
+        hour_2 = d_2.get(h)
+        h2 = (int(h-1))
+        hour_3 = d_5.get((int(str(h2)[1])))
+        minute_4 = d_3.get((int(str(m)[0]))*10)
+        minute_5 = d_3.get((int(str(m)[1])))
+        minute_2 = d_6.get(m)
+        print(f"{minute_4[0]} {minute_5[0]} минут{minute_2[0]} {hour_3[0]}")
+    elif ((h>=0 and h<=11) or (h>=21 and h<=24)) and ((m>=21 and m<=29) or (m>=31 and m<=39) or (m>=41 and m<=44)): # (часы  <12 и 21-23 /    минуты 21-29, 31-39, 41-44)
+        hour = d.get(h)
+        hour_2 = d_2.get(h)
+        h2 = (int(h+1))
+        hour_3 = d_5.get(h2)
+        minute_4 = d_3.get((int(str(m)[0]))*10)
+        minute_5 = d_3.get((int(str(m)[1])))
+        minute_2 = d_6.get(m)
+        print(f"{minute_4[0]} {minute_5[0]} минут{minute_2[0]} {hour_3[0]}")
+    elif (h>=12 and h<=20) and (m>=45 and m<=59): # (часы >=12 и до 20 вкл / минуты 45-59)
+        hour = d.get(h)
+        hour_2 = d_2.get(h)
+        h2 = (int(h-1))
+        hour_3 = d.get((int(str(h2)[1])))
+        m3 = (int(60-m))
+        minute_3 = d_8.get(m3)
+        minute_4 = d_7.get(m3)
+        print(f"без {minute_3[0]} минут{minute_4[0]} {hour_3[0]}")
+    elif ((h>=0 and h<=11) or (h>=21 and h<=24)) and (m>=45 and m<=59): # (часы <12 и 21-23  / минуты 45-59)
+        hour = d.get(h)
+        hour_2 = d_2.get(h)
+        h2 = (int(h+1))
+        hour_3 = d.get(h2)
+        m3 = (int(60-m))
+        minute_3 = d_8.get(m3)
+        minute_4 = d_7.get(m3)
+        print(f"без {minute_3[0]} минут{minute_4[0]} {hour_3[0]}")
+    else: print("error")
```

Comment:
it's a copy of the logic above, you could initially set up variables for hours and minutes (either from user or from datetime) first and then describe your logic only once